From: Julian Ospald <hasufell@gentoo.org>
Date: Sat Jun 30 20:48:23 UTC 2012
Subject: build system

use system fonts

make datadir and bindir paths modifiable
https://github.com/mana/mana/commit/02a7b2aa739cd0cf5a855f0a78b91d9fd5ad4385
don't overwrite system/environment CXXFLAGS
https://github.com/mana/mana/commit/2305ca095bee6b32c930f71b99d4c1205ceb561b
add option to use system enet headers
https://github.com/mana/mana/commit/d754a64e3839b3888d489046546195792e4ecc34

--- data/CMakeLists.txt.old	2011-08-02 12:15:38.520030381 +0200
+++ data/CMakeLists.txt	2011-08-02 12:15:44.971029615 +0200
@@ -1,6 +1,5 @@
 SET(DATA_DIR ${PKG_DATADIR}/data)
 
-ADD_SUBDIRECTORY(fonts)
 ADD_SUBDIRECTORY(graphics)
 ADD_SUBDIRECTORY(help)
 ADD_SUBDIRECTORY(icons)
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -42,9 +42,9 @@
     IF (NOT OSX AND NOT BEOS)
         OPTION(USE_X11 "Use X11 Clipboard functionality" ON)
     ENDIF ()
-    SET(PKG_DATADIR ${CMAKE_INSTALL_PREFIX}/share/mana)
+    SET(PKG_DATADIR ${CMAKE_INSTALL_PREFIX}/share/mana CACHE PATH "Mana datadir")
     SET(LOCALEDIR ${CMAKE_INSTALL_PREFIX}/share/locale)
-    SET(PKG_BINDIR ${CMAKE_INSTALL_PREFIX}/bin)
+    SET(PKG_BINDIR ${CMAKE_INSTALL_PREFIX}/bin CACHE PATH "Mana bindir")
 ENDIF (WIN32)
 
 ADD_SUBDIRECTORY(data)
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -12,7 +12,7 @@
 
 IF (CMAKE_COMPILER_IS_GNUCXX)
     # Help getting compilation warnings
-    SET(CMAKE_CXX_FLAGS "-Wall")
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
     IF (WIN32)
         # This includes enough debug information to get something useful
         # from Dr. Mingw while keeping binary size down. Almost useless
--- libs/enet/CMakeLists.txt
+++ libs/enet/CMakeLists.txt
@@ -1,3 +1,7 @@
+INCLUDE(FindPkgConfig)
+
+OPTION(WITH_BUNDLEDHEADERS "Use bundled enet headers" ON)
+
 INCLUDE(CheckFunctionExists)
 CHECK_FUNCTION_EXISTS(gethostbyaddr_r HAS_GETHOSTBYNAME_R)
 IF(HAS_GETHOSTBYNAME_R EQUAL 1)
@@ -49,7 +53,12 @@
   win32.c
   )
 
-INCLUDE_DIRECTORIES("include")
+IF (WITH_BUNDLEDHEADERS)
+	INCLUDE_DIRECTORIES("include")
+ELSE (WITH_BUNDLEDHEADERS)
+	PKG_CHECK_MODULES(ENET REQUIRED libenet)
+	INCLUDE_DIRECTORIES("${ENET_INCLUDEDIR}")
+ENDIF()
 
 ADD_LIBRARY(enet STATIC ${SRCS})
 IF (BEOS)
